synapseutils/sync.py

```diff
@@ -919,3 +933,27 @@ def _manifest_upload(syn, df):
         uploader.upload(items)
 
     return True
+
+
+def _check_file_name(df):
+    for idx, row in df.iterrows():
+        pattern = "^[`\\w \\-\\+\\.\\(\\)]{1,256}$"
+        if not row['name']:
+            directory_name = os.path.basename(row['path'])
+            df.loc[df.path == row['path'], 'name'] = row['name'] = directory_name
+            sys.stdout.write('The file name you assigned at path: %s is empty, so we set the store name as %s.\n'
+                             % (row['path'], directory_name))
+        if not re.match(pattern, row['name']):
+            raise ValueError('The file name on your local side is invalid to store on Synapse. Names may only contain:'
+                             'letters, numbers, spaces, underscores, hyphens, periods, plus signs, apostrophes,'
+                             'and parentheses')
+
+
+def _check_size_each_file(df):
+    for f in df.path:
+        if is_url(f):
+            continue
+        single_file_size = os.stat(os.path.expandvars(os.path.expanduser(f))).st_size
+        if single_file_size == 0:
+            print('\nAll the files uploaded cannot be 0 byte.')
+            raise ValueError('All the files uploaded cannot be 0 byte.')
```

Comment:
You don't need to print and raise an error, raising an error should be enough. But the message should have the file name in it. Each "File {} is empty, empty files cannot be uploaded to Synapse"